tests/h/h_api/bulk_api/model/command_test.py

```diff
@@ -114,16 +115,19 @@ def test_prepare_for_execute_pops_merge_query_from_config(self, user_command):
 
         assert config == {"another": True}
 
-    def test_prepare_for_execute_merges_queries(self, user_command, group_command):
-        # You don't get mixed command types, but it's helpful here as one has
-        # a query and the other doesn't
+    def test_prepare_for_execute_merges_queries(self, upsert_command):
+        query = deepcopy(upsert_command.body.meta["query"])
+        assert query
 
-        assert "groupid" in group_command.body.meta["query"]
-        assert "groupid" not in group_command.body.attributes
+        for key in query.keys():
+            assert key not in upsert_command.body.attributes
```

Comment:
Not sure how this was passing before. But basically when an upsert command is processed with 'merge_query' on, the query is copied into the attributes. 

The serialisation happens before this, but we were taking the commands as our `expected` after the process. This made it look like we were missing values, where the test was actually wrong.